### ISSUE TITLE
Inconsistency between exception handling (observed in node:events) behaviour with Node vs Bun

### ISSUE THREAD

**Issue by dominicporteous** [oven-sh/bun#1556]
### What version of Bun is running?

0.2.2 (fyi I have also verified this behaviour exists on canary builds 0.3.0)

### What platform is your computer?

Darwin 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:17:10 PDT 2022; root:xnu-8020.140.49~2/RELEASE_X86_64 x86_64 i386

### What steps can reproduce the bug?

I've noticed a difference in how Node and Bun (specifically the node:events polfil) handle (or don't handle, I guess) exceptions which are re-thrown from within an error handler.

Both Bun and Node seem to catch the error and print it out to console, but Node exits with an error code (which I think is the expected behaviour). I thought this might have been a difference between how bun and node handle unhandled exceptions/errors in general (legacy node would not exit), I have tested this back to Node v12 and the behaviour is consistent within Node - it will print the error to the console and exit with an error code (1)

I have created a repo here which reproduces the behaviour above: https://github.com/dominicporteous/bun-node-events

By running the test script within `./test` you can see the difference in how Bun and Node handle the unhandled error within the event emitter.

event.throws.js
```
import { EventEmitter } from 'node:events';
import { fileURLToPath } from 'node:url';
import process from 'process';

const main = (silent) => {

    const e = new EventEmitter();

    e.on('error', (err) => {

        silent ? '' : console.log('caught err, will rethrow', err)

        throw err

    })

    e.emit('error', new Error('Firing error'))

}

export default main

if (process.argv[1] === fileURLToPath(import.meta.url)) {
    main()
}
```

test
```
#/usr/bin/env bash

echo -e "\nRunning file..."
echo -e "\n===============================\n\n"
cat event.throws.js
echo -e "\n\n===============================\n\n"

echo "NODE_VERSION=$(node -v)"
echo "BUN_VERISON=$(bun -v)"

echo -e "\n\n"

BUN_EXIT="$(bun event.throws.js 1> /dev/null; echo "$?")"
NODE_EXIT="$(node event.throws.js 1> /dev/null; echo "$?")"

echo "BUN_EXIT=$BUN_EXIT"
echo "NODE_EXIT=$NODE_EXIT"
```

example output
```
NODE_VERSION=v18.8.0
BUN_VERISON=0.3.0

12 | 
13 |         throw err
14 | 
15 |     })
16 | 
17 |     e.emit('error', new Error('Firing error'))
                        ^
error: Firing error
      at /Users/dom/Github/bun-node-events/event.throws.js:17:20
      at /Users/dom/Github/bun-node-events/event.throws.js:24:4
node:internal/process/esm_loader:97
    internalBinding('errors').triggerUncaughtException(
                              ^

Error: Firing error
    at main (file:///Users/dom/Github/bun-node-events/event.throws.js:17:21)
    at file:///Users/dom/Github/bun-node-events/event.throws.js:24:5
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:528:24)
    at async loadESM (node:internal/process/esm_loader:91:5)
    at async handleMainPromise (node:internal/modules/run_main:65:12)

Node.js v18.8.0
BUN_EXIT=0
NODE_EXIT=1
```

### How often does it reproduce? Is there a required condition?

Seems to be consistent

### What is the expected behavior?

Expect both Node and Bun to catch the exception and print to console
Expect both Node and Bun to exit with a 1 error code

### What do you see instead?

Both Node/Bun seems to correctly catch the exception and print to console, but Bun will exit with a 0 exit code vs Node 1 exit code.

**Comment by dominicporteous** [comment]
FYI - I have observed the behaviour above in normal running (directly running the script with Node and Bun) and also when testing, to check if the call to main will actually re-throw the error (using throws from `node:assert`)

**Comment by paperclover** [comment]
this was fixed sometime in 0.6 or one of the few releases after that.
